camelot/handlers.py

```diff
@@ -25,21 +28,36 @@ class PDFHandler:
 
     Parameters
     ----------
-    filepath : str
-        Filepath or URL of the PDF file.
+    filepath : str | pathlib.Path, optional (default: None)
+        Filepath or URL of the PDF file. Required if file_bytes is not given
     pages : str, optional (default: '1')
         Comma-separated page numbers.
         Example: '1,3,4' or '1,4-end' or 'all'.
     password : str, optional (default: None)
         Password for decryption.
+    file_bytes : io.IOBase, optional (default: None)
+        A file-like stream. Required if filepath is not given
 
     """
 
-    def __init__(self, filepath: Union[StrByteType, Path], pages="1", password=None):
+    def __init__(self, filepath: Union[StrByteType, Path, None] = None, pages="1", password=None, file_bytes=None):
```

Comment:
StrByteType is already a Union and Unions cannot be nested. 

Is it an idea to introduce a custom type like this:

```python
FilePathType = Union[str, IO[Any], Path, None]`

def __init__(self, filepath: FilePathType = None, pages="1", password=None, file_bytes=None):
```